src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java

```diff
@@ -165,13 +211,37 @@ public Group getGroupById(Integer id) {
     return groupRep.findOne(id);
   }
 
+  /**
+   * Checks if group is valid. If group already exists, group is not valid. Method is used by
+   * addGroup method.
+   *
+   * @param group is a group we want to add.
+   * @return true if group is valid, or false if group is not valid.
+   */
   @Override
   public boolean isValid(Group group) {
     Group existed = groupRep.findByName(group.getName());
     return existed == null;
   }
 
   /**
+   * Checks if group's name is valid. Name is valid if the other groups don't have the same name.
+   * Method is used by updateGroup method.
+   *
+   * @param group is a group we want to update.
+   * @return true if group's name is valid, or false if group's name is not valid.
+   */
+  @Override
+  public boolean isValidGroupName(Group group) {
```

Comment:
In this case I don't think so, because we are returning a boolean value. This method provides validation test, that is why all logic has to be in the service layer. IMHO.